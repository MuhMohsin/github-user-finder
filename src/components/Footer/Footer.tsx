import * as React from "react"
import styled from "styled-components"
import { Link } from "gatsby"

interface Props {}

/** Style */
const Footer = styled.footer`
  margin: 30px auto 0;
  text-align: center;
  color: #bbb;
`

const Span = styled.span`
  color: #fabd22;
  border-bottom: 1px dotted #fabd22;
  text-decoration: none;
`

const FooterComp: React.FC<Props> = () => {
  return (
    <Footer>
      Build with <Span>React</Span> and <Span>Gatsby</Span> available on{" "}
      <Span as={Link} to="/">
        Github
      </Span>
      .
    </Footer>
  )
}

export default React.memo(FooterComp)
