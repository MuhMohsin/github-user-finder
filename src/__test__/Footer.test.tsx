import { configure } from "enzyme"
import EnzymeAdapter from "enzyme-adapter-react-16"
import configureStore from "redux-mock-store" //ES6 modules
import React from "react"
import { shallow, mount } from "enzyme"
import Footer from "../components/Footer/Footer"
import { Provider } from "react-redux"
import action from "../state/createStore"

const middlewares = []
const mockStore = configureStore(middlewares)

configure({ adapter: new EnzymeAdapter() })
// You would import the action from your codebase in a real scenario
describe("Navba test", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      startup: { complete: false },
    })
  })

  test("Should render true", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test("Should render true matching text", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>
    )
    expect(wrapper.text()).toBe(
      "Build with React and Gatsby available on Github."
    )
  })
})
