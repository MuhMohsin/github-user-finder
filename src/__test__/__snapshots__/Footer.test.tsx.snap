// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navba test Should render true 1`] = `ReactWrapper {}`;
